nautobot/core/management/commands/generate_performance_test_endpoints.py

```diff
@@ -0,0 +1,143 @@
+from django.core.management.base import BaseCommand
+from django.urls import get_resolver
+import yaml
+
+from nautobot.core.utils.lookup import get_model_for_view_name
+
+
+class Command(BaseCommand):
+    """
+    Example usage: `nautobot-server generate_performance_test_endpoints > endpoints.yml`
+    """
+
+    help = "List all relevant performance test url patterns in Nautobot Core"
+
+    def handle(self, *args, **options):
+        # Get the URL resolver
+        url_patterns = get_resolver().url_patterns
+
+        # Group the urls by app names
+        self.app_name_to_urls = {}
+        self.app_name_to_urls["endpoints"] = {}
+        # Fetch and store the urls by app names in the dictionary
+        self.fetch_urls(url_patterns)
+        with open("./endpoints.yml", "w") as outfile:
+            yaml.dump(self.app_name_to_urls, outfile, sort_keys=True)
+
+    def is_get_endpoint(self, view_name):
+        """
+        Check if the view is a GET endpoint
+        """
+        if view_name.endswith(("_list", "_notes", "_changelog")) or "_" not in view_name:
```

Comment:
This is very hacky, but introspecting the view class for allowed HTTP methods seems even more complex than I thought it would be because of the number of edge cases we need to handle between class-based views and UIViewSet-based views.

Reply:
Hacky but readable for now :-). Might check with @lampwins to see if we want to skip over some or all of the `_notes` and `_changelog` views since those are two generic views that are reused a bunch of times, rather than distinct views for each model.